server/cache.go

```diff
@@ -336,6 +348,18 @@ func (c *clusterInfo) putStoreLocked(store *storeInfo) error {
 	return nil
 }
 
+func (c *clusterInfo) blockStore(storeID uint64) {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
So you consider whether to use `defer` by checking the control flow every time you use a lock, and every time you change the code wrapping by the lock, you reconsider whether you should use `defer` again?